strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/pypi/PypiArtifactController.java

```diff
@@ -0,0 +1,351 @@
+import org.carlspring.strongbox.storage.validation.artifact.ArtifactCoordinatesValidationException;
+import org.carlspring.strongbox.web.LayoutRequestMapping;
+import org.carlspring.strongbox.web.RepositoryMapping;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.MediaType;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestPart;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import com.google.common.collect.Sets;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+/**
+ * Rest End Points for Pypi Artifacts requests.
+ * This controller will be Entry point for various pip commands.
+ * 
+ * @author ankit.tomar
+ * 
+ */
+@RestController
+@LayoutRequestMapping(PypiArtifactCoordinates.LAYOUT_NAME)
+public class PypiArtifactController extends BaseArtifactController
+{
+
+    private static final Set<String> VALID_ACTIONS = Sets.newHashSet("file_upload");
+
+    private static final Set<String> VALID_FILE_TYPES = Sets.newHashSet("sdist", "bdist_wheel");
+
+    @Inject
+    private RepositoryProviderRegistry repositoryProviderRegistry;
+
+    @ApiOperation(value = "This end point will be used to upload/deploy python package.")
+    @ApiResponses(value = { @ApiResponse(code = HttpURLConnection.HTTP_OK, message = "python package was deployed successfully."),
+                            @ApiResponse(code = HttpURLConnection.HTTP_INTERNAL_ERROR, message = "An error occurred while executing request."),
+                            @ApiResponse(code = HttpURLConnection.HTTP_UNAVAILABLE, message = "Service Unavailable.") })
+    @PreAuthorize("hasAuthority('ARTIFACTS_DEPLOY')")
+    @RequestMapping(path = "/{storageId}/{repositoryId}", method = RequestMethod.POST, consumes = MediaType.MULTIPART_FORM_DATA)
+    public ResponseEntity<String> uploadPackage(
+                                                @PathVariable(name = "storageId") String storageId,
```

Comment:
@anki2189 you can use `@RepositoryMapping` annotation to inject `Reposiotry` object like this:
```
@RepositoryMapping Repository repository
```

Reply:
Only storageId and repository was required so used @PathVariable. 
Will change it to @RepositoryMapping as suggested.